### DISCUSSION TITLE
Create Permission if not exists in HasPermission Trait

### DISCUSSION THREAD

**Discussion by marvin-wtt** [spatie/laravel-permission#2030] [Ideas]
When using wildcard permissions with the scheme `{resource}.{action}.{target}`, I'd find it more handy if the permission would be created automatically with `Permission#findOrCreate` instead of trowing an exception if the permission for the target does not exist yet. This allows the assignment of permissions directly, without having to check wherever it exists first. This makes it more intuitive in my opinion.
It could be combined with the config entry `enable_wildcard_permission`. 

Before creating a PR for this minor change, I'd like to hear some other opinions on this.

**Comment by erikn69** [comment]
Maybe it could be a breaking change

**Reply by marvin-wtt** [reply]
True. Adding a separate config key which defaults to false would avoid this. 
Any opinion on the change in general?

**Reply by erikn69** [reply]
No opinion, but the author doesn't really like making changes to the library because of the maintenance involved
You can try a PR, it might take some time for them to review it
